src/schema/server_settings.py

```diff
@@ -139,25 +139,73 @@ class Difficulty(str, Enum):
 ]
 
 
+def SettingField(
+        default: Optional[Any],
+        label: str,
+        description: str,
+        type: str,  # todo make enum
+        listof: Optional[str] = None,  # todo make enum
+        options: Optional[List[Dict]] = None,  # todo make inflated class
+        include_dynamic_options: Optional[str] = None,  # TODO enum with image-themes
+        required: Optional[bool] = None,
+        unused: Optional[bool] = None,
+        list_schema: Optional[List[Dict]] = None,  # how to represent this?
```

Comment:
Re: the list_schema comment -- Could it be another Pydantic class that's also using SettingField objects? That'd be pretty rad.

Reply:
Yeah I imagine we'd probably do the same kind of schema mechanism (maybe generalized to a mixin).  Will let that come in in a later PR.